libraries/AP_WheelEncoder/AP_WheelEncoder.cpp

```diff
@@ -155,15 +156,26 @@ void AP_WheelEncoder::init(void)
         return;
     }
     for (uint8_t i=0; i<WHEELENCODER_MAX_INSTANCES; i++) {
-#if CONFIG_HAL_BOARD == HAL_BOARD_CHIBIOS
         switch ((WheelEncoder_Type)_type[i].get()) {
+
+#if CONFIG_HAL_BOARD == HAL_BOARD_CHIBIOS
         case WheelEncoder_TYPE_QUADRATURE:
             drivers[i] = new AP_WheelEncoder_Quadrature(*this, i, state[i]);
             break;
+#endif
+
+#if CONFIG_HAL_BOARD == HAL_BOARD_SITL
+        case WheelEncoder_TYPE_SITL_QUADRATURE:
+            drivers[i] = new AP_WheelEncoder_SITL_Qaudrature(*this, i, state[i]);
+            break;
+#endif
         case WheelEncoder_TYPE_NONE:
             break;
+            
+        default:
```

Comment:
Please don't add defaults here - move the #ifs to within the case statements above.

Adding `default` like this means the compiler can no longer protect against all enumeration values being handled.